cfme/tests/webui/test_advanced_search.py

```diff
@@ -257,3 +163,179 @@ def test_filter_crud(param, appliance):
     else:
         if view.my_filters.is_displayed:
             assert not view.my_filters.navigation.has_item(filter_name), "Filter wasn't deleted!"
+
+
+_tests = [_advanced_search_button_displayed, _can_open_advanced_search, _filter_crud]
+
+
+def methodized(metafunc):
+    # TODO generalize for more tests with possibly different arguments
+    def func(self, param, appliance):
+        return metafunc(param, appliance)
+    return func
+
+
+def inject_tests(metaclass):
```

Comment:
This is pretty genius, but I'd agree with @lina-nikiforova, needs to be commented so that its clear that this is attaching test methods to the decorated classes